### PR TITLE
Added 7 Graph queries to APIM checklist

### PR DESCRIPTION
**Pull request by seenu433**
Added 7 Graph queries to APIM checklist

### CODE REVIEW

**Review by andredewes — COMMENTED** [review]
@seenu433 Shouldn't it be better for the recommendations that are only available in Premium instances that ARG queries filter only for sku == Premium?

For example, for customers that have Developer/Basic/Standard tiers they will be flagged in the recommendations about multi regional and AZ even though that's not possible.

**Comment by seenu433** [comment]
@andredewes - Unlike Advisor recommendations (for blast radius reasons), I would rather flag all instances in scope against compliance as we do not know the criticality of the workload.

**Comment by erjosito** [comment]
You probably need to use single quotes in the ARG queries, otherwise the JSON is not valid

File: checklists/apim_checklist.en.json [hunk 1]
```diff
@@ -139,6 +139,7 @@
             "waf": "Reliability",
             "guid": "beae759e-4ddb-4326-bf26-47f87d3454b6",
             "severity": "Medium",
+            "graph": "resources | where type == "microsoft.apimanagement/service" | extend compliant = ( sku.name == "Premium" and  isnotnull(properties.additionalLocations)) | distinct id, compliant",
```
Comment (erjosito) [inline checklists/apim_checklist.en.json:142]: Double quotes in the query string break the JSON syntax, please use single quotes

File: checklists/apim_checklist.en.json [hunk 2]
```diff
@@ -149,6 +150,7 @@
             "waf": "Reliability",
             "guid": "9c8d1664-dd9a-49d4-bd83-950af0af4044",
             "severity": "Medium",
+            "graph": "resources | where type == "microsoft.apimanagement/service" | extend compliant = ( sku.name == "Premium" and  isnotnull(zones) and sku.capacity >= 2 ) | distinct id, compliant",
```
Comment (erjosito) [inline checklists/apim_checklist.en.json:153]: Double quotes in the query string break the JSON syntax, please use single quotes

File: checklists/apim_checklist.en.json (hunk 1 above)
Comment (erjosito) [inline checklists/apim_checklist.en.json:142]: Double quotes in the query string break the JSON syntax, please use single quotes

File: checklists/apim_checklist.en.json [hunk 3]
```diff
@@ -209,6 +211,7 @@
             "text": "Configure autoscaling to scale out the number of instances when the load increases",
             "guid": "bb5f356b-3daf-47a2-a9ee-867a8100bbd5",
             "severity": "Medium",
+            "graph": "resources | where type == "microsoft.apimanagement/service" | join kind = leftouter (resources | where type == "microsoft.insights/autoscalesettings" | extend targetResourceUri = tostring(properties.targetResourceUri)) on $left.id == $right.targetResourceUri | extend compliant = (sku.name == "Premium" and isnotempty(targetResourceUri) and properties1.enabled == true) | distinct id, compliant",
```
Comment (erjosito) [inline checklists/apim_checklist.en.json:214]: Double quotes in the query string break the JSON syntax, please use single quotes

File: checklists/apim_checklist.en.json [hunk 4]
```diff
@@ -279,6 +282,7 @@
             "text": "Deploy the service within a Virtual Network (VNet)",
             "guid": "cd45c90e-7690-4753-930b-bf290c69c074",
             "severity": "Medium",
+            "graph": "resources | where type == "microsoft.apimanagement/service" | extend compliant = (isnotnull(properties.virtualNetworkConfiguration)) | distinct id, compliant",
```
Comment (erjosito) [inline checklists/apim_checklist.en.json:285]: Double quotes in the query string break the JSON syntax, please use single quotes

File: checklists/apim_checklist.en.json [hunk 5]
```diff
@@ -299,6 +303,7 @@
             "text": "Deploy Private Endpoints to filter incoming traffic when APIM is not deployed to a VNet.",
             "guid": "67437a28-2721-4a2c-becd-caa54c8237a5",
             "severity": "Medium",
+            "graph": "resources | where type == "microsoft.apimanagement/service" | extend compliant = (properties.virtualNetworkType == "None" and isnotnull(properties.privateEndpointConnections)) | distinct id, compliant",
```
Comment (erjosito) [inline checklists/apim_checklist.en.json:306]: Double quotes in the query string break the JSON syntax, please use single quotes

File: checklists/apim_checklist.en.json [hunk 6]
```diff
@@ -309,6 +314,7 @@
             "text": "Disable Public Network Access",
             "guid": "d698adbd-3288-44cb-b10a-9b572da395ae",
             "severity": "High",
+            "graph": "resources | where type == "microsoft.apimanagement/service" | extend compliant = (properties.virtualNetworkType == "Internal") | distinct id, compliant",
```
Comment (erjosito) [inline checklists/apim_checklist.en.json:317]: Double quotes in the query string break the JSON syntax, please use single quotes

File: checklists/apim_checklist.en.json [hunk 7]
```diff
@@ -419,6 +425,7 @@
             "text": "Use managed identities to authenticate to other Azure resources whenever possible",
             "guid": "791abd8b-7706-4e31-9569-afefde724be3",
             "severity": "Medium",
+            "graph": "resources | where type == "microsoft.apimanagement/service" | extend compliant = (isnotnull(identity)) | distinct id, compliant",
```
Comment (erjosito) [inline checklists/apim_checklist.en.json:428]: Double quotes in the query string break the JSON syntax, please use single quotes

**Review by erjosito — CHANGES_REQUESTED** [review]

**Comment by seenu433** [comment]
@erjosito -comments addressed.

**Review by erjosito — APPROVED** [review]

**Pull request merged**